### PR TITLE
Explicitly point out where should we set the random seed

### LINKED ISSUE #3460: Missing details in the document for the random seed
## 📚 Documentation

Even though in the [documentation](https://pytorch-lightning.readthedocs.io/en/latest/multi_gpu.html), lightning very thoughtfully reminds readers of setting the random seed when using distributed data parallel. However, it does not mention where should we set them. Based on my experience on PyTorch, the seed can be set after calling the `torch.distributed.init_process_group`. For lightning, this call is handled by the trainer, so where should we set the random seed? One thought from me is that we set the seed in the `__init__()` of our LightningModule, any other suggestions? What would be the best practice for lightning? 

Also, I believe that the `seed_everything()` function is called in the wrong place on the lightning's official [example](https://github.com/PyTorchLightning/pytorch-lightning/blob/114af8ba9fc42fcf7053fa06299fbe4aecab8a06/pl_examples/basic_examples/sync_bn.py).

### PR DESCRIPTION
**Pull request by DKandrew**
## What does this PR do?

Refactor the document, let it to be more friendly to newcomers in PyTorch framework 

Fixes #3460

# Before submitting

- [ x ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ x ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ x ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ x ] Did you make sure to update the documentation with your changes?
- [ x ] Did you write any new necessary tests? 
- [ x ] Did you verify new and existing tests pass locally with your changes?
- [ x ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/multi_gpu.rst [hunk 1]
```diff
@@ -260,7 +260,7 @@ Distributed Data Parallel
 
 3. Each process inits the model.
 
-.. note:: Make sure to set the random seed so that each model initializes with the same weights.
+.. note:: Make sure to set the random seed before the instantiation of a `Trainer()` so that each model initializes with the same weights.
```
Comment (ydcjeff) [inline docs/source/multi_gpu.rst:263]:
```suggestion
.. note:: Make sure to set the random seed before the instantiation of a ``Trainer()`` so that each model initializes with the same weights.
```
